spec/Dynamic-Validator.md

```diff
@@ -0,0 +1,144 @@
+
+## Nomination
+Any account, that is not banned, can nominate itself.
+The account becomes a candidate when the sum of the deposit is more than **MIN_DEPOSIT**.
+The nomination expires after **NOMINATION_EXPIRATION**; the account who wants to remain a candidate must nominate itself before the previous nomination expires.
+The deposit reverts to the account when it becomes a normal account.
+
+### Minimum Deposit
+TBD
+
+## Delegation
+The stakeholders have the right to choose validators as must as their shares.
+It's called a delegation.
+And the stakeholders who have been delegated called delegators.
+The delegation is valid only when the delegatee is not a normal account or not banned.
+The delegated stakes are returned when the account becomes a normal account or banned.
+
+## Election
+The process that elects validators of a term is called the election.
+The election selects validators as the following rule.
+<<<<<<<
+1. **MAX_NUM_OF_VALIDATORS** accounts in order of having received many delegations.
+2. Pick **MIN_NUM_OF_VALIDATORS** accounts; they become validators.
+3. In the rest of them, drops the accounts having received less than **MIN_CCS_RATE_TO_BE_VALIDATOR**.
+4. The remains become validators.
+=======
+The initial *MIN_DELEGATION* is **MIN_CCS_RATE_TO_BE_VALIDATOR**.
+1. Pick accounts in order of having received many delegations.
+2. Drops the accounts having received more than *MIN_DELEGATION*
+3. If the number of accounts is more than **MIN_NUM_OF_VALIDATORS** selects top **MAX_NUM_OF_VALIDATORS** accounts.
+4. Otherwise, repeat from step 1 after changing *MIN_DELEGATION* to half.
+>>>>>>>
+
+## Voting Power
+Each elected validators has different voting power.
+The voting power is based on the delegation that the validator has received at the election.
+The block is valid only if the sum of voting power is more than 2/3 of the total delegations that elected validators received.
+
+## Validator Reward
+The block proposer gets the expression fee of the blocks at the end of a term.
+Validators can get the reward after **WITHDRAW_DELAY** terms.
+However, the proposers cannot get all reward if he is not loyal to their duty.
+The reward is decreased according to the rate of the blocks the validator misses to sign.
+TBD: The rate of decreasing.
+
+## Punishment for Validators
+### Downtime
+The validator who doesn't produce blocks is jailed for a while.
+The jailed account cannot be a candidate during **CUSTODY_PERIOD**.
+*SELF_NOMINATE* transaction of the account is rejected.
+Actually, this is not a punishment.
+It is to give validators time to fix the nodes they manage.
+The jailed account can nominate itself again after **CUSTODY_PERIOD**.
+
+### Disloyal Validators
+CodeChain gives a penalty to the validator who doesn't participate in signing the blocks proposed by another node.
+See [Validator Reward](#Validator-Reward) for more information.
+
+### Double Vote
+CodeChain bans the account who double voted.
```

Comment:
Would you describe more about what banning means here?

Reply:
done